x/concentrated-liquidity/incentives.go

```diff
@@ -359,6 +367,18 @@ func (k Keeper) updateUptimeAccumulatorsToNow(ctx sdk.Context, poolId uint64) er
 		return err
 	}
 
+	// Claim and clear the balancer full range shares from the current pool's uptime accumulators.
+	// This is to avoid having to update accumulators every time the canonical balancer pool changes state.
+	// Even though this exposes CL LPs to getting immediately diluted by a large Balancer position, this would
+	// require a lot of capital to be tied up in a two week bond, which is a viable tradeoff given the relative
+	// simplicity of this approach.
```

Comment:
There's still a slight difference – adding to the CL pool (SFS or not) would still distribute incentives prior to updating pool liquidity so only future incentives would get diluted. In this case, historical incentives since the last liquidity update would get diluted as well.